File: RaspberryPi/python_demo/09_active_buzzer/09_active_buzzer.py
# !/usr/bin/env python3
# -*- coding: utf-8 -*-
# file name: active_buzzer.py
# version: 1.0
# author: Neal Shen
# ------------------------------------------------------------------
import RPi.GPIO as GPIO
import time
makerobo_BuzzerPin = 11    # active buzzer pin


# gpio inilization
def makerobo_setup():
    GPIO.setmode(GPIO.BOARD)
    GPIO.setwarnings(False)
    GPIO.setup(makerobo_BuzzerPin, GPIO.OUT)
    GPIO.output(makerobo_BuzzerPin, GPIO.HIGH) # set buzzer high level, disable buzzer

# enable buzzer
def makerobo_buzzer_on():
    GPIO.output(makerobo_BuzzerPin, GPIO.LOW) # set buzzer low level, enable buzzer


# disable buzzer
def makerobo_buzzer_off():
    GPIO.output(makerobo_BuzzerPin, GPIO.HIGH) # set buzzer high level, disable buzzer


# control buzzer
def makerobo_beep(x):
    makerobo_buzzer_on()    # enable buzzer control
    time.sleep(x)
    makerobo_buzzer_off()   # disable buzzer control
    time.sleep(x)


# loop function
def makerobo_loop():
    while True:
        makerobo_beep(0.5)  # delay time 500ms


# release source
def makerobo_destroy():
    GPIO.output(makerobo_BuzzerPin, GPIO.HIGH)
    GPIO.cleanup()


if __name__ == '__main__':
    makerobo_setup()
    try:
        makerobo_loop()
    except KeyboardInterrupt:
        makerobo_destroy()

File: RaspberryPi/cplus_demo/09_active_buzzer/09_active_buzzer.c
/**********************************************
 * ---------------active buzzer---------------
 * file name: active_buzzer.c
 * version: 1.0
 * author: Neal Shen
 *********************************************/
#include <wiringPi.h>
#include <stdio.h>

#define makerobo_BuzzerPin		0			// active buzzer's pin

int main(void)
{
	// if the initialization is failed, print the infomation
	// on the screen of raspberrypi
	if (wiringPiSetup() == -1)
	{
		printf("setup wiringPi failed !");
		return -1;
	}

	// set buzzer pin mode is output
	pinMode(makerobo_BuzzerPin, OUTPUT);

	while (1) {
		// set pin low level, enable buzzer
		digitalWrite(makerobo_BuzzerPin, LOW);
		delay(100);		//delay 100ms
		// set pin high level, disable buzzer
		digitalWrite(makerobo_BuzzerPin, HIGH);
		delay(100);		//delay 100ms
	}

	return 0;
}